docs/source/conf.py

```diff
@@ -162,6 +191,48 @@
 # Customize the html_theme
 html_theme_options = {'navigation_depth': 3}
 
+# Template for generating the components.rst file
```

Comment:
Include this in a template named components.rst which gets populated later on, using something similar to what you've got, or a templating engine like [jinga](http://zetcode.com/python/jinja/)?

Reply:
I think it's only worth splitting out if we use a templating engine like jinga, otherwise I'd need to split it out into 4 files (I have 4 variables containing rst code). Sphinx depends on jinja2 (used for the HTML templates) anyway so I guess I'll use it! Then I can put everything in one .rst template file and switch code blocks in/out, have for loops, etc.

Reply:
This should now be done (once it's built online)

I've also added the "LINKS" caption as done in labscript-suite/runmanager#89 (which is a good way to tell if you are looking at the new version).

Will roll out to the other docs if you are happy

Reply:
Wow! Amazing job. See comments and rpanderson/labscript-suite@91fdaad061bd755697dd2083745e1042100e78dc regarding simplifications by using yaml.